src/menus/icon-menu.jsx

```diff
@@ -228,10 +230,13 @@ const IconMenu = React.createClass({
   },
 
   _handleItemTouchTap(event, child) {
-    const isKeyboard = Events.isKeyboard(event);
-    this.timerCloseId = setTimeout(() => {
-      this.close(isKeyboard ? 'enter' : 'itemTap', isKeyboard);
-    }, this.props.touchTapCloseDelay);
+
+    if (this.props.touchTapCloseDelay > 0) {
```

Comment:
I believe that if `touchTapCloseDelay === 0` this block should be called synchronously. It's not the case right now.